src/modules/modules.py

```diff
@@ -126,87 +161,45 @@ def forward(self, sent, task, reset=True):
         # Skip computing this if it won't be used.
-        if self.sep_embs_for_skip:
-            if isinstance(self._text_field_embedder, BertEmbedderModule):
-                task_sent_embs = self._text_field_embedder(sent, task._classifier_name,
-                                                           is_pair_task=is_pair_task)
-
-            else:
-                task_sent_embs = self._text_field_embedder(sent, task._classifier_name)
-            task_sent_embs = self._highway_layer(task_sent_embs)
-        else:
-            task_sent_embs = None
+        if self.sep_elmo_embs_for_skip:
+            task_sent_embs = self._highway_layer(
+                self._text_field_embedder(
+                    sent, task._classifier_name))
+            task_sent_embs = self._dropout(task_sent_embs)
 
         # Make sure we're embedding /something/
         assert (sent_embs is not None) or (task_sent_embs is not None)
 
         if self._cove_layer is not None:
-            # Slightly wasteful as this repeats the GloVe lookup internally,
-            # but this allows CoVe to be used alongside other embedding models
-            # if we want to.
-            sent_lens = torch.ne(
-                sent['words'],
-                self.pad_idx).long().sum(
-                dim=-1).data
-            # CoVe doesn't use <SOS> or <EOS>, so strip these before running.
-            # Note that we need to also drop the last column so that CoVe returns
-            # the right shape. If all inputs have <EOS> then this will be the
-            # only thing clipped.
-            sent_cove_embs_raw = self._cove_layer(sent['words'][:, 1:-1],
-                                                  sent_lens - 2)
-            pad_col = torch.zeros(sent_cove_embs_raw.size()[0], 1,
-                                  sent_cove_embs_raw.size()[2],
-                                  dtype=sent_cove_embs_raw.dtype,
-                                  device=sent_cove_embs_raw.device)
-            sent_cove_embs = torch.cat(
-                [pad_col, sent_cove_embs_raw, pad_col], dim=1)
-            if sent_embs is not None:
-                sent_embs = torch.cat([sent_embs, sent_cove_embs], dim=-1)
-            if task_sent_embs is not None:
-                task_sent_embs = torch.cat(
-                    [task_sent_embs, sent_cove_embs], dim=-1)
-
-        if sent_embs is not None:
-            sent_embs = self._dropout(sent_embs)
-        if task_sent_embs is not None:
-            task_sent_embs = self._dropout(task_sent_embs)
+            task_sent_embs, sent_embs = get_cove_layer_embeddings(sent, sent_embs, task_sent_embs)
 
         # The rest of the model
         sent_mask = util.get_text_field_mask(sent).float()
         sent_lstm_mask = sent_mask if self._mask_lstms else None
-        if sent_embs is not None:
-            sent_enc = self._phrase_layer(sent_embs, sent_lstm_mask)
+        if self.sent_enc_type != 'null':
```

Comment:
this seems like a lot more work (of passing around `sent_enc_type` for a relatively small benefit. Why is this change necessary?

Reply:
e.g. we could still use the check ` isinstance(self._phrase_layer, NullPhraseLayer)`, I think?

Reply:
isinstance(self._phrase_layer, NullPhraseLayer is not clear as to how the NullPhraaseLayer is triggered. At essence, it's triggered when sent_enc = null, so I passed it in for code clarity